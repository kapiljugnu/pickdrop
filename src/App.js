import React, { Component } from 'react';
import { withScriptjs } from 'react-google-maps';
import Map from './Map';
import LocationInput from './Location/Input';
import './App.css';
import { getPath } from './Api/action';

export class App extends Component {

  state = { path: [] }

  onLocationSubmit = (pick, drop) => {
    if (pick && drop) {
      getPath({ pick, drop }).then(({ data: { path = [] } }) => this.setState({ path }));
    }
  }

  render() {
    return (
      <div className="container">
        <div className="form">
          <LocationInput
            onSubmit={this.onLocationSubmit} />
        </div>
        <Map
          path={this.state.path}
          containerElement={<div className="map" />}
          mapElement={<div style={{ height: `100%` }} />}
        />
      </div>
    );
  }
}

export default withScriptjs(App);