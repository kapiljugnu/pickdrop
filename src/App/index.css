.container {
    display: flex;
    flex-direction: row;
    position: relative;
    min-height: 100vh;
}

.form {
    flex: 1;
    order: 1;
    margin: 0;
    padding: 0;
}

.map{
    flex: 3;
    order: 2;
    position: relative;
    min-height: 100%;
}