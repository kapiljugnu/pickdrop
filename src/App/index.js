import React, { Component } from 'react';
import { withScriptjs } from 'react-google-maps';
import Map from '../Map';
import LocationInput from '../Location/Input';
import './index.css';
import { getPath } from '../Api/action';
import Direction from '../Map/Direction';
import { SERVER_ERROR, INPUT_ERROR } from '../constants/messages';

export class App extends Component {

  state = { path: [], hasError: false, message: '' }

  onLocationSubmit = (pick, drop) => {
    this.setState({ hasError: false, message: '' })
    if (pick && drop) {
      getPath({ pick, drop })
        .then(({ data: { path = [] } }) => this.setState({ path }))
        .catch((error) => { this.setState({ hasError: true, message: SERVER_ERROR }) });
    } else {
      this.setState({ hasError: true, message: INPUT_ERROR })
    }
  }

  onReset = () => {
    this.setState({ path: [], hasError: false, message: '' });
  }

  render() {
    return (
      <Direction path={this.state.path}>
        {
          ({ directions, distance, duration }) => <div className="container">
            <div className="form">
              <LocationInput distance={distance.toString()} duration={duration.toString()} onSubmit={this.onLocationSubmit} onReset={this.onReset} />
              {this.state.hasError && <div className="error">{this.state.message}</div>}
            </div>
            <Map
              directions={directions}
              containerElement={<div className="map" />}
              mapElement={<div style={{ height: `100%` }} />}
            />
          </div>
        }
      </Direction>
    );
  }
}

export default withScriptjs(App);