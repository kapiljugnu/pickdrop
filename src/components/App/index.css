.container {
    display: flex;
    flex-direction: row;
    position: relative;
    min-height: 100vh;
}

.form {
    flex: 1;
    order: 1;
    margin: 0;
    padding: 0;
}

.map{
    flex: 3;
    order: 2;
    position: relative;
    min-height: 100%;
}

.error {
    color: red;
    margin: 12px 0;
}

.loader {
    border: 5px solid #f3f3f3;
    border-top: 5px solid #555;
    border-radius: 50%;
    width: 50px;
    height: 50px;
    animation: spin 2s linear infinite;
  }

.form .loader {
  margin-top: 3px;
  width:20px;
  height: 20px;
 }

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }