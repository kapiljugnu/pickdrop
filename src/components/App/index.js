import React, { Component } from 'react';
import { withScriptjs } from 'react-google-maps';
import Map from '../Map';
import Location from '../Location';
import './index.css';
import getPath from '../../services/getPath';
import Direction from '../Map/Direction';
import { SERVER_ERROR, INPUT_ERROR } from '../../constants/messages';

export class App extends Component {

  state = { path: [], hasError: false, message: '', isFetching: false }

  onLocationSubmit = (pick, drop) => {
    if (!pick || !drop) {
      this.setState({ hasError: true, message: INPUT_ERROR, isFetching: false });
      return;
    }
    this.setState({ hasError: false, message: '', isFetching: true })
    getPath({ pick, drop })
      .then(({ data: { path = [] } }) => this.setState({ path }))
      .catch((error) => { this.setState({ hasError: true, message: SERVER_ERROR }) })
      .finally(() => { this.setState({ isFetching: false }) })
  }

  onReset = () => {
    this.setState({ path: [], hasError: false, message: '' });
  }

  render() {
    return (
      <Direction path={this.state.path}>
        {
          ({ directions, distance, duration }) => <div className="container">
            <div className="form">
              <Location distance={distance.toString()} duration={duration.toString()} onSubmit={this.onLocationSubmit} onReset={this.onReset} />
              {this.state.isFetching && <div className="loader" />}
              {this.state.hasError && <div className="error">{this.state.message}</div>}
            </div>
            <Map
              directions={directions}
              containerElement={<div className="map" />}
              mapElement={<div style={{ height: `100%` }} />}
            />
          </div>
        }
      </Direction>
    );
  }
}

export default withScriptjs(App);