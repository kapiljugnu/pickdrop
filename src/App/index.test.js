import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './index';

it('renders without crashing', () => {
  const app = mount(<App
    loadingElement={<div style={{ height: `100%` }} />}
    googleMapURL='googleMapURL'
  />).setState({ loadingState: 'LOADED' });
  expect(toJson(app)).toMatchSnapshot();
});
