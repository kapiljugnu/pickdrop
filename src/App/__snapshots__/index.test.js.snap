// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<withScriptjs(App)
  googleMapURL="googleMapURL"
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
>
  <App>
    <Direction
      path={Array []}
    >
      <div
        className="container"
      >
        <div
          className="form"
        >
          <Input
            distance=""
            duration=""
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <form>
              <label
                htmlFor="pick"
              >
                Starting Location
              </label>
              <StandaloneSearchBox
                onPlacesChanged={[Function]}
              >
                <input
                  id="pick"
                  placeholder="Starting Location"
                  type="text"
                />
              </StandaloneSearchBox>
              <label
                htmlFor="drop"
              >
                Drop-off point
              </label>
              <StandaloneSearchBox
                onPlacesChanged={[Function]}
              >
                <input
                  id="drop"
                  placeholder="Drop-off point"
                  type="text"
                />
              </StandaloneSearchBox>
              <br />
              <input
                onClick={[Function]}
                type="submit"
                value="Submit"
              />
              <input
                onClick={[Function]}
                type="button"
                value="Reset"
              />
            </form>
          </Input>
        </div>
        <withGoogleMap(Map)
          containerElement={
            <div
              className="map"
            />
          }
          directions={null}
          mapElement={
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          }
        >
          <div
            className="map"
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
            <div />
          </div>
        </withGoogleMap(Map)>
      </div>
    </Direction>
  </App>
</withScriptjs(App)>
`;
